tests/src/assembler_rv64i_tests.cpp

```diff
@@ -76,6 +77,147 @@ TEST_CASE("LD", "[rv64i]") {
     REQUIRE(value == 0xFFFFB783);
 }
 
+TEST_CASE("LI64", "[rv64i]") {
+    // Up to 8 instructions can be generated
+    std::array<uint32_t, 8> vals{};
+    Assembler as(reinterpret_cast<uint8_t*>(vals.data()), sizeof(vals));
+
+    const auto compare_vals = [&vals](int n...) {
```

Comment:
Please don't use C varargs.

This can be done like:

```cpp
const auto compare_vals = [&vals]<typename... Args>(const Args&... args) {
    static_assert(sizeof...(args) <= vals.size());

    size_t i = 0;
    for (const auto arg : {args...}) {
        REQUIRE(vals[i] == arg);
        i++;
    }
};
```